src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -2492,7 +2493,7 @@ private void initFileMetadata() throws SnowflakeSQLException {
           } else if (file.isDirectory()) {
             logger.debug("Not a file, but directory: {}", sourceFile);
 
-            throw new SnowflakeSQLException(
```

Comment:
ah, good catch.

I think the signature of SnowflakeSQLLoggedException should be updated to have the session to the first one to avoid the conflict, such as
```
public SnowflakeSQLLoggedException(String SQLState, int vendorCode, SFSession session, Object... params);
```
with
```
public SnowflakeSQLException(String sqlState, int vendorCode, Object... params) {
```.

Reply:
That's a good idea. I can make that change.